test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs

```diff
@@ -93,68 +93,344 @@ await foreach (Page<Product> page in allPages.AsPages())
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            Operation<Product> result = await new DPGClient(Key, host, null, nestedScopeSuppression).LroValueAsync(WaitUntil.Completed, "raw");
+            diagnosticListener.AssertAndRemoveScope("DPGClient.LroValue");
+
+            if (nestedScopeSuppression)
+            {
+                CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            }
+
+            Assert.AreEqual("raw", result.Value.Received.ToString());
+        }, ignoreScenario: true);
+
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public Task RawLROWaitUntilCompletedOperation(bool nestedScopeSuppression) => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            Operation result = await new DPGClient(Key, host, null, nestedScopeSuppression).LroValueAsync(WaitUntil.Completed, "raw");
+            diagnosticListener.AssertAndRemoveScope("DPGClient.LroValue");
+
+            if (nestedScopeSuppression)
+            {
+                CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            }
+
+            JsonData responseBody = JsonData.FromBytes(result.GetRawResponse().Content.ToMemory());
+            Assert.AreEqual("raw", (string)responseBody["received"]);
+        }, ignoreScenario: true);
+
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public Task HandwrittenModelLro(bool nestedScopeSuppression) => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            Operation<Product> lro = await new DPGClient(Key, host, null, nestedScopeSuppression).LroValueAsync(WaitUntil.Started, "model");
+            diagnosticListener.AssertAndRemoveScope("DPGClient.LroValue");
+            if (!nestedScopeSuppression)
+            {
+                diagnosticListener.AssertAndRemoveScope("DPGClient.Lro");
+            }
             CollectionAssert.IsEmpty(diagnosticListener.Scopes);
 
             await lro.WaitForCompletionAsync();
             diagnosticListener.AssertAndRemoveScope("DPGClient.LroValue.WaitForCompletion");
-            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            if (nestedScopeSuppression)
+            {
+                CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+            }
             Assert.AreEqual("model", $"{lro.Value.Received}");
         });
 
         [Test]
-        public Task HandwrittenModelLro_ManualIteration() => Test(async (host) =>
+        [TestCase(true)]
+        [TestCase(false)]
+        public Task HandwrittenModelLroResponseOperationOfT(bool nestedScopeSuppression) => Test(async (host) =>
```

Comment:
This is a lower-priority comment, but given the duplication of complex logic across these tests, I'm wondering if there's a way to implement the scope-checking once, and reuse it across the protocol and convenience methods, rather than re-implementing it by hand?